util/variant.h

```diff
@@ -11,6 +11,11 @@
 
 namespace util {
 
+template<typename... T, typename... Ts>
+constexpr bool holds_any_of(std::variant<Ts...> const &v) noexcept {
+    return (std::holds_alternative<T>(v) || ...);
```

Comment:
Can you add some unit tests to variant_test.cpp so this is tested even if wasm/... changes?